File: backend/gatorsched_api/schemas/features/employee_schedule.py
from pydantic import BaseModel


class EmployeeShiftSummary(BaseModel):
    fromTime: str
    toTime: str
    longLabel: str
    dateLabel: str
    shiftHours: float


class EmployeeShift(BaseModel):
    key: str
    shortLabel: str
    timeRange: str
    isoDate: str
    summary: EmployeeShiftSummary


class EmployeeScheduleResponse(BaseModel):
    id: str
    color: str
    weekLabel: str
    totalHours: float
    schedule: list[EmployeeShift]

File: frontend/src/services/employeeSchedule.ts
import { API_BASE } from './apiConfig';

export type EmployeeShiftSummary = {
  fromTime: string;
  toTime: string;
  longLabel: string;
  dateLabel: string;
  shiftHours: number;
};

export type EmployeeShift = {
  key: string;
  shortLabel: string;
  timeRange: string;
  isoDate: string;
  summary: EmployeeShiftSummary;
};

export type EmployeeScheduleResponse = {
  id: string;
  color: string;
  weekLabel: string;
  totalHours: number;
  schedule: EmployeeShift[];
};

export async function getEmployeeSchedule(
  viewerId: string,
  weekStart: string,
): Promise<EmployeeScheduleResponse> {
  const res = await fetch(
    `${API_BASE}/employee/schedule?viewer_id=${encodeURIComponent(viewerId)}&week_start=${encodeURIComponent(weekStart)}`,
  );

  if (!res.ok) {
    throw new Error('Failed to retrieve Schedule');
  }

  return await res.json();
}
